packages/library-base/src/commonMain/kotlin/io/realm/RealmObject.kt

```diff
@@ -88,7 +97,7 @@ internal fun RealmObject.hasSameObjectKey(other: RealmObject?): Boolean {
 public fun RealmObject.isValid(): Boolean {
     return if (isManaged()) {
         val internalObject = this as RealmObjectInternal
-        val ptr = internalObject.`$realm$ObjectPointer`
+        val ptr = internalObject.`$realm$objectReference`!!.objectPointer
```

Comment:
I don't have the full overview, but we have these nullchecks all over the place, which looks a bit glaring to me and indicates there is some state or type we haven't nailed just yet. It would be nice if we could somehow get rid of this.

Reply:
Yes, claus also suggested it. Now I do the check implicitly when extracting the `ObjectReference`.